topoflow/components/met_base.py

```diff
@@ -1291,10 +1312,10 @@ def update_P_rain(self):
         #-------------------------------------------------
         P_rain = self.P * (self.T_air > self.T_rain_snow)
         
-        if (np.ndim( self.P_rain ) == 0):
+        if ((np.ndim( self.P_rain ) == 0) & (self.T_air_type.lower() == 'scalar')): # Why is it self.P_rain and not P_rain? LB

```

Comment:
Ah, I see what you're saying about P_rain vs. self.P_rain -- thanks! 

Regarding the T_air_type clause, Scott's comments suggest that it _should_ be possible if P_rain is a scalar for T_air to be scalar or a grid (e.g. if the user used the temperature lapse rate utility I created as part of the PR to turn a temperature time series into a grid stack), and I think this clause is necessary to make that possible. Otherwise it tries to fill the scalar P_rain with the gridded self.P_rain and throws an error.